src/fu-common-cab.c

```diff
@@ -187,6 +187,8 @@ fu_common_store_from_cab_file (XbBuilder *builder, GCabCabinet *cabinet,
 	g_autoptr(GError) error_local = NULL;
 	g_autoptr(XbBuilderSource) source = xb_builder_source_new ();
 
+	/* rewrite to be under a components root */
+	xb_builder_source_set_prefix (source, "components");
```

Comment:
I can't believe I didn't think of that. It's exactly what the `xb_builder_source_set_prefix` call is useful for. Maybe add one empty line under this between the other comment, but I'm really nitpicking here.

Reply:
Yeah it had me scratching my head too, it looked like it fit the bill perfectly.